ext/QuantumToolboxCairoMakieExt.jl

```
@@ -139,6 +140,84 @@ function _plot_wigner(
+- `unit_y_range::Bool`: Set y-axis limits [0, 1] or not, default is `true`.
+- `kwargs...`: Additional keyword arguments to pass to the plotting function. 
+
+# Returns
+- `fig`: The figure object.
+- `ax`: The axis object.
+- `hm`: Either the heatmap or surface object, depending on the projection.
+
+!!! note "Import library first"
+    [`CairoMakie`](https://github.com/MakieOrg/Makie.jl/tree/master/CairoMakie) must first be imported before using this function.
+
+!!! warning "Beware of type-stability!"
+    If you want to keep type stability, it is recommended to use `Val(:two_dim)` and `Val(:three_dim)` instead of `:two_dim` and `:three_dim`, respectively. Also, specify the library as `Val(:CairoMakie)` See [this link](https://docs.julialang.org/en/v1/manual/performance-tips/#man-performance-value-type) and the [related Section](@ref doc:Type-Stability) about type stability for more details.
+"""
+function QuantumToolbox.plot_fock_distribution(
+    library::Val{:CairoMakie},
+    ρ::QuantumObject{SType};
+    fock_numbers::Union{Nothing,AbstractVector} = nothing,
+    unit_y_range::Bool = true,
+    location::Union{GridPosition,Nothing} = nothing,
+    kwargs...,
+) where {SType<:Union{BraQuantumObject,KetQuantumObject,OperatorQuantumObject}}
+    return _plot_fock_distribution(
+        library,
+        ρ;
+        fock_numbers = fock_numbers,
+        unit_y_range = unit_y_range,
+        location = location,
+        kwargs...,
+    )
+end
+
+function _plot_fock_distribution(
+    ::Val{:CairoMakie},
+    ρ::QuantumObject{SType};
+    fock_numbers::Union{Nothing,AbstractVector} = nothing,
+    unit_y_range::Bool = true,
+    location::Union{GridPosition,Nothing} = nothing,
+    kwargs...,
+) where {SType<:Union{BraQuantumObject,KetQuantumObject,OperatorQuantumObject}}
+    ρ = ket2dm(ρ)
+    D = prod(ρ.dims)
+    (real(tr(ρ)) > 1) && (@warn "The input ρ should be normalized.")
```

But also when it is smaller than one we should raise an error, right? Is it better to use `isapprox(real(tr(ρ)), 1, atol=...)`?

---

Done